mailbox/store/src/main/java/org/apache/james/mailbox/store/search/comparator/SentDateComparator.java

```diff
@@ -72,15 +76,19 @@ static String sanitizeDateStringHeaderValue(String value) {
 
     @Override
     public int compare(MailboxMessage o1, MailboxMessage o2) {
-        Instant date1 = getSentDate(o1);
-        Instant date2 = getSentDate(o2);
-        return date1.compareTo(date2);
+        return parseSentDate(o1).compareTo(parseSentDate(o2));
     }
     
-    private Instant getSentDate(MailboxMessage message) {
-        final String value = getHeaderValue("Date", message);
-        return toISODate(value)
-            .map(ZonedDateTime::toInstant)
-            .orElse(message.getInternalDate().toInstant());
-    }
+    private Date parseSentDate(MailboxMessage message) {
```

Comment:
Let me help you improve previously existing code to avoid the all message mime4J parsing